pkg/gameservers/controller.go

```diff
@@ -520,10 +520,13 @@ func (c *Controller) syncDevelopmentGameServer(ctx context.Context, gs *agonesv1
 		return gs, nil
 	}
 
-	if !(gs.Status.State == agonesv1.GameServerStateReady) {
-		c.loggerForGameServer(gs).Debug("GS is a development game server and will not be managed by Agones.")
+	// Only move from Creating -> Ready. Other manual state changes are up to the end user.
+	// We also don't want to move from Allocated -> Ready every time someone allocates a GameServer.
+	if !(gs.Status.State == agonesv1.GameServerStateCreating) {
```

Comment:
Is there a reason that this isn't `if gs.Status.State != agonesv1.GameServerStateCreating { ... }`? The way it's currently written is difficult for my brain to parse.

Reply:
That's a really good point. I was just copying the original format, but yeah - I'll make this change.